src/overlays/actors/ovl_Boss_03/z_boss_03.c

```diff
@@ -41,157 +269,2320 @@ const ActorInit Boss_03_InitVars = {
+
+        xDiff = sp50.x - player->actor.world.pos.x;
+        zDiff = sp50.z - player->actor.world.pos.z;
+
+        if (sqrtf(SQ(xDiff) + SQ(zDiff)) < (2.0f * phi_f2)) {
+            Math_ApproachS(&this->jawZRot, 0x3200, 2, 0x1800);
+            this->unk_278 = 25.0f;
+            this->unk_27C = 5.0f;
+            this->skelAnime.playSpeed = 2.5f;
+        }
+
+        // Near enough to Player?
+        if (sqrtf(SQ(xDiff) + SQ(zDiff)) < phi_f2) {
+            Boss03_SetupCatchPlayer(this, globalCtx, sp43);
+
+            if (sp43 != 0) {
+                Actor_Spawn(&globalCtx->actorCtx, globalCtx, ACTOR_EN_WATER_EFFECT, player->actor.world.pos.x,
+                            this->waterHeight, player->actor.world.pos.z, 0, 0, 0x78, ENWATEREFFECT_777);
+                Boss03_PlayUnderwaterSfx(&this->actor.projectedPos, NA_SE_EN_KONB_SINK_OLD);
+            }
+
+            Boss03_PlayUnderwaterSfx(&this->actor.projectedPos, NA_SE_EN_UTSUBO_EAT);
+        }
+    }
+
+    Boss03_SpawnDust(this, globalCtx);
+}
+
+void Boss03_SetupCatchPlayer(Boss03* this, GlobalContext* globalCtx, u8 arg2) {
+    this->actionFunc = Boss03_CatchPlayer;
+    Animation_MorphToLoop(&this->skelAnime, &gGyorgFastSwimmingAnim, -15.0f);
+    this->workTimer[WORK_TIMER_CURRENT_ACTION] = 100;
+    this->unk_2C4 = 0.0f;
+    this->unk_2B8 = 0.0f;
+    this->unk_242 = arg2;
+}
+
+void Boss03_CatchPlayer(Boss03* this, GlobalContext* globalCtx) {
+    Player* player = GET_PLAYER(globalCtx);
+    f32 xDiff;
+    f32 yDiff;
+    f32 zDiff;
+    s32 pad;
+
+    this->skelAnime.playSpeed = 2.0f;
+    this->unk_27C = 2.0f;
+    this->unk_276 = 0x1000;
+    this->unk_2BD = true;
+    this->unk_278 = 15.0f;
+    this->actor.flags &= ~ACTOR_FLAG_1;
+
+    SkelAnime_Update(&this->skelAnime);
+
+    xDiff = player->actor.world.pos.x - this->actor.world.pos.x;
+    yDiff = player->actor.world.pos.y - this->actor.world.pos.y;
+    zDiff = player->actor.world.pos.z - this->actor.world.pos.z;
+
+    Math_ApproachS(&this->actor.world.rot.x, Math_FAtan2F(sqrtf(SQ(xDiff) + SQ(zDiff)), -yDiff), 0xA, this->unk_274);
+    Math_ApproachS(&this->bodyYRot,
+                   Math_SmoothStepToS(&this->actor.world.rot.y, Math_FAtan2F(zDiff, xDiff), 0xA, this->unk_274, 0) *
+                       -0.5f,
+                   5, 0x100);
+    Math_ApproachS(&this->unk_274, this->unk_276, 1, 0x100);
+    Math_ApproachF(&this->actor.speedXZ, this->unk_278, 1.0f, this->unk_27C);
+    Math_ApproachF(&this->unk_260, __sinf(this->skelAnime.curFrame * (M_PI / 5.0f)) * 10.0f * 0.01f, 0.5f, 1.0f);
+    Actor_MoveWithoutGravityReverse(&this->actor);
+    Math_ApproachS(&this->actor.shape.rot.x, this->actor.world.rot.x, 2, this->unk_274 * 2);
+    Math_ApproachS(&this->actor.shape.rot.y, this->actor.world.rot.y, 2, this->unk_274 * 2);
+
+    // If either (Player is standing on ground && Player is above water) or (WORK_TIMER_CURRENT_ACTION timer runs out)
+    // -> Stop trying to catch Player
```

Comment:
I think you need to decide if you're going to use this style or the one from the previous comment. I'd prefer the previous one, I think, it looks less like a Python function or something.